nameko/timer.py

```diff
@@ -49,26 +52,29 @@ def kill(self):
 
     def _run(self):
         """ Runs the interval loop. """
-
-        sleep_time = self.interval
+        sleep_time = 0 if self.eager else self.interval
 
         while True:
             # sleep for `sleep_time`, unless `should_stop` fires, in which
             # case we leave the while loop and stop entirely
-            with Timeout(sleep_time, exception=False):
```

Comment:
how come you are changing this? the old behaviour is explained in the comment just above and doesn't work with the replacement.

basically, if the service is stopped while the timer is sleeping, we don't want to wait until it's done sleeping before tearing down the timer entrypoint